### PR TITLE
Decoupling/component/post

### PR DESCRIPTION
**Pull request by cheesea3**
src/components/Post/Post.container.tsx (export Post)
src/components/Post/Post.component.tsx (Export PostComponent, rendered inside Post container)
Created src/components/Post/Post.emptyState.tsx (Export EmptyPostState)

Moved SlideUpMenu to post component.
Decoupled the post component entirely alongside the empty state component. Notes have been added for future documentation and design has been made with future implementation in mind. Types are enforced and the components are robust. Conditionally, emptyPostState is shown when the posts array is empty, otherwise it will load the posts. 

Functions, state, and various logic remains in the parent component while being passed down to the children component's up to three levels down. Passed down function props not only account for the parameters being accessed while following naming convention, but also the expected return type whether that be void or a return value specifically such as a boolean. Since state is intended to be handled in the uppermost parent component, each function explicitly uses void.

### CODE REVIEW

File: src/components/SlideUpMenu.tsx [hunk 1]
```diff
@@ -2,15 +2,12 @@ import { useCallback, useEffect } from 'react'
...
-
-  return true
-}
+//  currently not functional UNTIL search button is merged to seperate area of concerns
+//  for this to work, css needs to be included.
 
 interface MoreOptionsProps {
   isOwner?: boolean
-  setMenuVisibility(value: boolean): boolean
+  setMenuVisibility(isMenuOpen: boolean): void
```
Comment (sterlingwalker) [inline src/components/SlideUpMenu.tsx:10]:
```suggestion
  setMenuVisibility: (isMenuOpen: boolean) => void
```

File: src/pages/Timeline.tsx [hunk 2]
```diff
@@ -19,86 +24,100 @@ const Timeline: React.FC = () => {
...
-    '
-      >
-        {printHeader()}
-      </IonHeader>
+      <Toolbar></Toolbar>
       <IonContent
         id='main'
         fullscreen={true}
         color='light'
+        class=''
```
Comment (sterlingwalker) [inline src/pages/Timeline.tsx:78]:
```suggestion
```

Unused

File: src/pages/Timeline.tsx [hunk 3]
```diff
@@ -19,86 +24,100 @@ const Timeline: React.FC = () => {
...
+      postId: 5,
+      username: 'another',
+      isOwner: false,
+      likes: 2,
+    },
+  ]
+
   return (
-    <IonPage id='main'>
+    <IonPage id='main' className="">
```
Comment (sterlingwalker) [inline src/pages/Timeline.tsx:71]:
```suggestion
    <IonPage id='main'>
```

Unused

**Review by sterlingwalker — APPROVED** [review]
Just wanna say that this was a very impressive refactor, well done! 💯 I think this is super organized and easy to how all of the pieces come together. One last thing I would have said was to run prettier but since you have another PR coming down the pipeline we can wait on that to avoid conflicts lol.

File: src/components/MoreOptions.tsx [hunk 4]
```diff
@@ -11,8 +11,8 @@ export const disableContentScrollY = (contentEl: HTMLElement): boolean => {
 
 interface MoreOptionsProps {
   isOwner?: boolean
-  setPostID(postId: number): number
-  setMenuVisibility(value: boolean): boolean
+  setPostID(postId: number): void
+  setMenuVisibility(value: boolean): void
```
[the remark this answers is not in the document]
Comment (cheesea3) [reply src/components/MoreOptions.tsx:15]: I agree! Thank you for the reminder and you're right! This is the right way to go about it and allows us to maintain a ubiquitous design throughout the other components. Thank you for the heads up, i'll be sure to approach it through this from hear on out.

[the remark this answers is not in the document]
Comment (cheesea3) [reply src/components/MoreOptions.tsx:15]: I agree! Thank you for the reminder and you're right! This is the right way to go about it and allows us to maintain a ubiquitous design throughout the other components. Thank you for the heads up, i'll be sure to approach it through this from here on out.

File: src/components/Post/Post.container.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+const Post = (props: postObject) => {
+  /* allows us to trim content of the post if it exceeds 140 characters. 
+  The remaining content is stored in a temporary variable that is revealed if the user clicks 'show more'. */
+  const trimText = (text: string) => {
+    // configurable by us to define the max length until the read more button appears.
+    const maxLength = 140
+    // cleans up the string
+    const originalContent = text.trim()
```
[the remark this answers is not in the document]
Comment (cheesea3) [reply src/components/Post/Post.container.tsx:33]: Something to definitely consider, I think we should definitely clean up formatting before entering it into the database. You're right, it does make sense to clean prior to insert so then when it doesn't need to be cleaned when read. Lets keep this implementation for now and revisit when we couple the front-end to the backend. 💯 Agreed!

File: src/components/Post/Post.container.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+          {isPostCollapsed ? originalContent : content + '...'}
+        </span>
+        {originalContent.length > maxLength ? (
+          <button
+            className='text-base font-medium text-blue-400 capitalize transition active:opacity-50 hover:text-blue-400 focus:text-blue-500 focus:outline-none'
+            onClick={(e) => readMorePostHandler(e)}
+          >
+            {isPostCollapsed ? 'show less' : 'show more'}
+          </button>
+        ) : null}
```
[the remark this answers is not in the document]
Comment (cheesea3) [reply src/components/Post/Post.container.tsx]: You're right! I overcomplicated it there, didn't notice that i could avoid the "else" clause in the condition when writing ternary. Committed the changes and included the link.

**Pull request merged**